CRM/Custommosaico/Upgrader.php

```diff
@@ -1,29 +1,107 @@
-    civicrm_api3('OptionGroup', 'create', ['name' => 'custommosaico_brand_fonts', 'title' => E::ts('Brand Fonts')]);
+    $this->createOptionGroup(FontLoaderPlugin::OPTION_GROUP);
   }
 
   public function uninstall() {
-    // Delete custommosaico_brand_fonts option group and its values.
+    $this->deleteOptionGroupAndValues(FontLoaderPlugin::OPTION_GROUP);
+  }
+
+  public function onEnable() {
+    $this->alterOptionGroup('enable', FontLoaderPlugin::OPTION_GROUP);
+  }
+
+  public function onDisable() {
+    $this->alterOptionGroup('disable', FontLoaderPlugin::OPTION_GROUP);
+  }
+
+  /**
+   * Alters option group
+   *
+   * @param $action
+   * @param $name
+   */
+  private function alterOptionGroup($action, $name) {
+    $optionGroupId = $this->getOptionGroupId($name);
+
+    $alterOptionGroup = [];
+
+    switch ($action) {
+      case 'enable':
+        $alterOptionGroup = [
+          'option_group_id' => $optionGroupId,
+          'api.OptionGroup.create' => [
+            'id' => '$value.id',
+            'is_active' => 1,
+          ],
+        ];
+        break;
+
+      case 'disable':
+        $alterOptionGroup = [
+          'option_group_id' => $optionGroupId,
+          'api.OptionGroup.create' => [
+            'id' => '$value.id',
+            'is_active' => 0,
+          ],
+        ];
+        break;
+
+      case 'uninstall':
+        $alterOptionGroup = [
+          'option_group_id' => $optionGroupId,
+          'api.OptionGroup.delete' => ['id' => '$value.id'],
+        ];
+        break;
+    }
+
+    civicrm_api3('OptionGroup', 'get', $alterOptionGroup);
+  }
+
+  private function getOptionGroupId($name) {
     try {
-      $optionGroupId = civicrm_api3('OptionGroup', 'getvalue', ['return' => 'id', 'name' => 'custommosaico_brand_fonts']);
-      $optionValues = civicrm_api3('OptionValue', 'get', ['option_group_id' => $optionGroupId, 'options' => ['limit' => 0]]);
-      foreach ($optionValues['values'] as $optionValue) {
-        civicrm_api3('OptionValue', 'delete', ['id' => $optionValue['id']]);
-      }
-      civicrm_api3('OptionGroup', 'delete', ['id' => $optionGroupId]);
+      $result = civicrm_api3('OptionGroup', 'getvalue', ['return' => 'id', 'name' => $name]);
+      return $result;
     }
     catch (\CiviCRM_API3_Exception $ex) {
-      // Ignore exception.
+      return FALSE;
+    }
+
+  }
+
+  public function createOptionGroup($name) {
+    $optionGroupId = $this->getOptionGroupid($name);
+    if (!empty($optionGroupId)) {
+      return;
+    }
+
+    civicrm_api3('OptionGroup', 'create',
+      [
+        'name' => $name,
+        'title' => E::ts('Brand Fonts'),
+      ]);
+  }
+
+  public function deleteOptionGroupAndValues($name) {
+    $optionGroupId = $this->getOptionGroupid($name);
+    if (empty($optionGroupId)) {
+      return;
+    }
+
+    $optionValues = civicrm_api3('OptionValue', 'get', ['option_group_id' => $optionGroupId, 'options' => ['limit' => 0]]);
```

Comment:
my idea is that you don't need to get the option values and delete them at all, you just want to do this:

```
civicrm_api3('OptionValue', 'delete', [
  'id' => $optionGroupId,
]);
```

and CiviCRM will automatically delete the option values.

Reply:
noted.